### PR TITLE
added simple inference script for binary classifier

### PR DESCRIPTION
**Pull request by ericlyfan**
We ran into issues that stemmed from the fact that the model was trained on cropped bounding boxes of the target class but ran inference on full resized images, causing a train/inference mismatch and significant amount of false positives. 

Fixed by adding a sliding window approach (128, 256 px). Also noticed potential issues with the data imbalance, so I balanced the training data to just the "target" class + one other class (instead of the full 19 classes), this produces very decent results.

### CODE REVIEW

File: src/common/binary_classifier_inference.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+import torch
+from binary_classifier import BinaryClassifier
+from PIL import Image, ImageDraw
+from torchvision import transforms
+
+FOLDER = "data/VOC2012/VOC2012_test/JPEGImages"
+MODEL_PATH = "checkpoints/best_model.pth"
+THRESHOLD = 0.9
+OUTPUT_PATH = "results/collage.jpg"
+WINDOW_SIZES = [256, 512]
```
Comment (Ansh-Gandhi) [inline src/common/binary_classifier_inference.py:16]: Should this window size be 128 x 256?

File: src/common/voc_dataset.py [hunk 2]
```diff
@@ -76,81 +77,78 @@ def __init__(
...
-        print(
-            f"Loaded {len(self.samples)} object instances from {len(self.image_ids)} images"
-        )
+        label_counts = Counter(s["class_name"] for s in self.samples)
+        print(f"Loaded {len(self.samples)} samples: {dict(label_counts)}")
 
     def _parse_annotations(self) -> None:
-        """Parse all XML annotations and create sample list."""
+        """Parse annotations - only include target class and one negative class."""
+        TARGET_CLASS_IDX = 14  # person
```
Comment (Ansh-Gandhi) [inline src/common/voc_dataset.py]: I am not 100% sure on this but there is a chance the labels may have to be 0 and 1 for the classification due to pytorch expecting that instead.

**Comment by Ansh-Gandhi** [comment]
Hey Eric, overall really good stuff! I left a few comments.

**Review by Ansh-Gandhi — CHANGES_REQUESTED** [review]

File: src/common/voc_dataset.py (hunk 2 above)
Comment (ericlyfan) [reply src/common/voc_dataset.py]: This should be okay because I convert to 0/1 labels in the training loop, here for reference: https://github.com/UBCAgroBot/target-object-localization/blob/1dcfa34e9c6f7ec67f7693d69044fc6cbe50a160/src/training.py#L141-L143

Comment (Ansh-Gandhi) [reply src/common/voc_dataset.py]: Sounds good, I missed that. Thanks Eric.
